src/databricks/labs/ucx/azure/access.py

```diff
@@ -130,116 +145,210 @@ def _map_storage(self, storage: StorageAccount) -> ValuesView[StoragePermissionM
-        uber_principal: PrincipalSecret,
-        inventory_database: str,
+        policy_id: str,
+        principal_client_id: str,
+        principal_secret_identifier: str,
+        storage_accounts: list[StorageAccount],
     ):
+        """Revert the cluster policy."""
+        try:
+            policy = self._installation.load(Policy, filename="policy-backup.json")
+        except NotFound:
+            try:
+                policy = self._ws.cluster_policies.get(policy_id)
+            except (InvalidParameterValue, NotFound):
+                return  # No policy to revert
+        policy_definition = policy.definition
+        if policy_definition is not None:
+            policy_definition = self._remove_service_principal_configuration_from_cluster_policy_definition(
+                policy_definition,
+                principal_client_id,
+                principal_secret_identifier,
+                storage_accounts,
+            )
+        if policy.name is not None:
+            self._ws.cluster_policies.edit(policy_id, policy.name, definition=policy_definition)
 
-        warehouse_config = self._ws.warehouses.get_workspace_warehouse_config()
-        sql_dac = warehouse_config.data_access_config
-        if sql_dac is None:
-            sql_dac = []
+    def _create_storage_account_data_access_configuration_pairs(
+        self, principal_client_id: str, principal_secret_identifier: str, storage: StorageAccount
+    ) -> list[EndpointConfPair]:
+        """Create the data access configuration pairs to access the storage"""
         tenant_id = self._azurerm.tenant_id()
         endpoint = f"https://login.microsoftonline.com/{tenant_id}/oauth2/token"
-        for storage in storage_account_info:
-            sql_dac.extend(
-                [
-                    EndpointConfPair(
-                        f"spark_conf.fs.azure.account.oauth2.client.id.{storage.name}.dfs.core.windows.net",
-                        uber_principal.client.client_id,
-                    ),
-                    EndpointConfPair(
-                        f"spark_conf.fs.azure.account.oauth.provider.type.{storage.name}.dfs.core.windows.net",
-                        "org.apache.hadoop.fs.azurebfs.oauth2.ClientCredsTokenProvider",
-                    ),
-                    EndpointConfPair(
-                        f"spark_conf.fs.azure.account.oauth2.client.endpoint.{storage.name}.dfs.core.windows.net",
-                        endpoint,
-                    ),
-                    EndpointConfPair(
-                        f"spark_conf.fs.azure.account.auth.type.{storage.name}.dfs.core.windows.net", "OAuth"
-                    ),
-                    EndpointConfPair(
-                        f"spark_conf.fs.azure.account.oauth2.client.secret.{storage.name}.dfs.core.windows.net",
-                        "{{secrets/" + inventory_database + "/uber_principal_secret}}",
-                    ),
-                ]
+        return [
```

Comment:
can you use `_create_service_principal_cluster_policy_configuration_pairs` method output? it's duplicated code, per se :)